nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetTCP.java

```diff
@@ -0,0 +1,668 @@
+                    } catch (Exception backupException) {
+
+                        log.error("Caught exception trying to connect to backup server: ", new Object[]{backupServer, ex});
+                        throw new ProcessException(String.format("Caught exception trying to create thread to process " +
+                                "data from backup server: %s", backupServer), backupException);
+                    }
+                } else {
+                    final String connectionMsg = connectedToBackup.get() ? String.format("Already connected to backup %s", backupServer) :
+                            "No backup server configured to connect to";
+                    log.info(connectionMsg);
+                    throw new ProcessException(String.format("Caught exception trying to create thread to process data from host: %s", hostAndPort[0]), ex);
+                }
+            }
+
+            getLogger().info("Created thread to process data from host: {}", new Object[]{hostAndPort[0]});
+            final Future socketReceiverFuture = executorService.submit(socketRecveiverThread);
+            socketToFuture.put(socketRecveiverThread, socketReceiverFuture);
+
+
+        }
+
+    }
+
+    protected SocketRecveiverThread createSocketRecveiverThread(int rcvBufferSize, boolean keepAlive, int connectionTimeout, int connectionRetryCount, String[] hostAndPort) {
+        SocketRecveiverThread socketRecveiverThread;
+        socketRecveiverThread = new SocketRecveiverThread(hostAndPort[0],
+                Integer.parseInt(hostAndPort[1]), backupServer, keepAlive, rcvBufferSize, connectionTimeout, connectionRetryCount);
+        return socketRecveiverThread;
+    }
+
+    @OnStopped
+    public void tearDown() throws ProcessException {
+        try {
+            for (Map.Entry<SocketRecveiverThread, Future> socketAndFuture : socketToFuture.entrySet()) {
+                socketAndFuture.getKey().stopProcessing();
+                socketAndFuture.getValue().cancel(true);
+
+            }
+            if (null != executorService) {
+                executorService.shutdown();
+                if (!executorService.awaitTermination(5, TimeUnit.SECONDS)) {
+                    executorService.shutdownNow();
+
+                    if (!executorService.awaitTermination(5, TimeUnit.SECONDS))
+                        log.error("Executor service for receiver thread did not terminate");
+                }
+            }
+        } catch (final Exception e) {
+            throw new ProcessException(e);
+        }
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        try {
+            final StringBuilder messages = new StringBuilder();
+            if (socketMessagesReceived.size() >= batchSize) {
+                for (int i = 0; i < batchSize; i++) {
+                    messages.append(socketMessagesReceived.poll(100, TimeUnit.MILLISECONDS));
```

Comment:
This can return null and we don't want that in the outgoing flowfile as a string.